### PR TITLE
Heet nodal support nonlinearity

### PR DESCRIPTION
**Pull request by heetrojivadiya**
# Description

Added all nonlinearity types for nodal support. There are still bug for assigning some attributes but I believe that they are from WS core and after fix all assignment will work fine.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit Tests
- [ ] Attached examples

**Test Configuration**:
* RFEM / RSTAB version: RFEM 6.03.0005
* Python version: 3.11

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: RFEM/enums.py [hunk 1]
```diff
@@ -2558,3 +2558,29 @@ class NodalSupportDiagramType(Enum):
...
+class SupportPartialActivityAlongType(Enum):
+    '''
+    Support Partial Activity Along Type Enumeration
+    '''
+    PARTIAL_ACTIVITY_TYPE_COMPLETE, PARTIAL_ACTIVITY_TYPE_FAILURE, PARTIAL_ACTIVITY_TYPE_FAILURE_FROM_FORCE, PARTIAL_ACTIVITY_TYPE_FIXED, \
+    PARTIAL_ACTIVITY_TYPE_YIELDING_FROM_FORCE = range(5)
+
+class SupportPartialActivityAroundType(Enum):
+    '''
+    Support Partial Activity Along Type Enumeration
```
Comment (aliciaknauer) [inline RFEM/enums.py:2571]: Hi,

In the class SupportPartialActivityAroundType, it should be 'around' in the comment.

File: RFEM/TypesForNodes/nodalSupport.py [hunk 2]
```diff
@@ -120,12 +120,12 @@ def Nonlinearity(no: int = 1,
                      nodes: str = "1",
                      coordinate_system: int = 1,
                      spring_constant: list = [inf, inf, inf, 0.0, 0.0, inf],
-                     spring_x_nonlinearity: list = [NodalSupportNonlinearity.NONLINEARITY_TYPE_NONE, [None], [None]],
```
Comment (OndraMichal) [inline RFEM/TypesForNodes/nodalSupport.py:123]: Why is this still list or list of lists as descriptions says? All 6 _nonlinearity parameters should be just enum. Make corresponding changes to the description also.

**Review by OndraMichal — CHANGES_REQUESTED** [review]

File: RFEM/TypesForNodes/nodalSupport.py (hunk 2 above)
Comment (heetrojivadiya) [reply RFEM/TypesForNodes/nodalSupport.py:123]: If the nonlinearity enumeration is partial_activity, diagram or friction_direction then it must be defined with additional list inside main list. I have created unittest for it where I have added example of other nonlinearity type with their additional parameters as list of lists. I also wrote complete description in docstrings when we should to define additional lists and what are the parameters for the same. 

image

File: RFEM/enums.py (hunk 1 above)
Comment (heetrojivadiya) [reply RFEM/enums.py:2571]: updated

File: RFEM/TypesForNodes/nodalSupport.py (hunk 2 above)
Comment (OndraMichal) [reply RFEM/TypesForNodes/nodalSupport.py:123]: ok

**Review by OndraMichal — APPROVED** [review]

**Pull request merged**